tracecontext/src/test/java/brave/propagation/tracecontext/TraceparentFormatTest.java

```diff
@@ -131,254 +113,249 @@
+  @Test void parseTraceparentFormat_newer_version_ignores_extra_fields() {
     assertThat(parseTraceparentFormat("10-" + traceIdHigh + traceId + "-" + spanId + "-00-fobaly"))
-      .isEqualToComparingFieldByField(TraceContext.newBuilder()
-        .traceIdHigh(Long.parseUnsignedLong(traceIdHigh, 16))
-        .traceId(Long.parseUnsignedLong(traceId, 16))
-        .spanId(Long.parseUnsignedLong(spanId, 16))
-        .sampled(false).build()
-      );
+      .usingRecursiveComparison().isEqualTo(TraceContext.newBuilder()
+      .traceIdHigh(Long.parseUnsignedLong(traceIdHigh, 16))
+      .traceId(Long.parseUnsignedLong(traceId, 16))
+      .spanId(Long.parseUnsignedLong(spanId, 16))
+      .sampled(false).build()
+    );
   }
 
-  @Test public void parseTraceparentFormat_newer_version_ignores_extra_flags() {
+  @Test void parseTraceparentFormat_newer_version_ignores_extra_flags() {
     assertThat(parseTraceparentFormat("10-" + traceIdHigh + traceId + "-" + spanId + "-ff"))
-      .isEqualToComparingFieldByField(TraceContext.newBuilder()
-        .traceIdHigh(Long.parseUnsignedLong(traceIdHigh, 16))
-        .traceId(Long.parseUnsignedLong(traceId, 16))
-        .spanId(Long.parseUnsignedLong(spanId, 16))
-        .sampled(true).build()
-      );
+      .usingRecursiveComparison().isEqualTo(TraceContext.newBuilder()
+      .traceIdHigh(Long.parseUnsignedLong(traceIdHigh, 16))
+      .traceId(Long.parseUnsignedLong(traceId, 16))
+      .spanId(Long.parseUnsignedLong(spanId, 16))
+      .sampled(true).build()
+    );
   }
 
   /** for example, parsing inside tracestate */
-  @Test public void parseTraceparentFormat_middleOfString() {
+  @Test void parseTraceparentFormat_middleOfString() {
     String input = "tc=00-" + traceIdHigh + traceId + "-" + spanId + "-01,";
     assertThat(parseTraceparentFormat(input, 3, input.length() - 1))
-      .isEqualToComparingFieldByField(TraceContext.newBuilder()
-        .traceIdHigh(Long.parseUnsignedLong(traceIdHigh, 16))
-        .traceId(Long.parseUnsignedLong(traceId, 16))
-        .spanId(Long.parseUnsignedLong(spanId, 16))
-        .sampled(true).build()
-      );
+      .usingRecursiveComparison().isEqualTo(TraceContext.newBuilder()
+      .traceIdHigh(Long.parseUnsignedLong(traceIdHigh, 16))
+      .traceId(Long.parseUnsignedLong(traceId, 16))
+      .spanId(Long.parseUnsignedLong(spanId, 16))
+      .sampled(true).build()
+    );
   }
 
-  @Test public void parseTraceparentFormat_middleOfString_incorrectIndex() {
+  @Test void parseTraceparentFormat_middleOfString_incorrectIndex() {
     String input = "tc=00-" + traceIdHigh + traceId + "-" + spanId + "-00,";
-    assertThat(parseTraceparentFormat(input, 0, 12))
-      .isNull(); // instead of raising exception
+    try (MockedStatic<Platform> mb = mockStatic(Platform.class)) {
+      mb.when(Platform::get).thenReturn(platform);
+
+      assertThat(parseTraceparentFormat(input, 0, 12))
```

Comment:
didn't complicate the more reused assertion as there's only one offset based test at the moment